src/util/serviceSelector.ts

```diff
@@ -0,0 +1,106 @@
+import {  ProcessorRequest, RouterRequestData } from "../types/index";
+import { MiscService } from "../services/misc.service";
+import { INTEGRATION_SERVICE } from "../routes/utils/constants";
+import CDKV1ServiceDestination from "../services/destination/cdkV1Integration.destination.service";
+import CDKV2ServiceDestination from "../services/destination/cdkV2Integration.destination.service";
+import IntegrationServiceDestination from "../interfaces/IntegrationDestinationService";
+import NativeIntegrationServiceDestination from "../services/destination/nativentegration.destination.service";
+import IntegrationServiceSource from "../interfaces/IntegrationSourceService";
+import NativeIntegrationServiceSource from "../services/source/nativeIntegration.source.service";
+
+export class ServiceSelector {
+  private static sourceHandlerMap: Map<string, any> = new Map();
+  private static destHandlerMap: Map<string, any> = new Map();
+  private static serviceMap: Map<string, any> = new Map();
+
+  private static isCdkDestination(destinationDefinitionConfig: Object) {
+    return !!destinationDefinitionConfig?.["cdkEnabled"];
+  }
+
+  private static isCdkV2Destination(destinationDefinitionConfig: Object) {
+    return !!destinationDefinitionConfig?.["cdkV2Enabled"];
+  }
+
+  private static fetchCachedService(serviceType: string) {
+    let service: any;
+    if (this.serviceMap.has(serviceType)) {
+      service = this.serviceMap.get(serviceType);
```

Comment:
`return  this.serviceMap.get(serviceType);` to avoid else case and nesting for other case

Reply:
We would want to store the services if not present right?